Update build to remove UTF-8 chars.

bfMega opened the pull request.

diegocr committed:
Removing UTF-8 characters as required by our deployment script...

diegocr committed:
Revert to the previous (1.4.3) of Autolinker.tldRegex...

bfMega committed:
Merge base

bfMega committed:
Formatting update for mega.

diegocr reviewed (changes requested):
Thanks Ben!

Under https://github.com/meganz/Autolinker.js/commit/e63dd8c0016e503e17b05f08ff97c2ba5417bf1f i took the chance to switch the build procedure to generate ES2018-compliant files instead.

Would you mind adding into `scripts/build.ts` a function that programmatically changes the UTF-8 characters, as a mean to ease further maintenance / upgrades in the long run, please? If you do find any trouble with this just sync the changes and apply manually, tho.

bfMega committed:
Merge upstream

bfMega committed:
Update build to remove UTF-8 chars

bfMega commented:
Removed some of the problematic characters in the rest of the code + Updated the tldRegex generating script to replace the UTF-8 characters. Also changed the rollup config to match the filename we use.

diegocr commented on scripts/update-tld-regex.ts:
```diff
@@ -16,7 +16,15 @@ export async function updateTldRegex() {
 
-    const tldRegex = domainsToRegex(tldsFile.data);
+    let tldRegex = domainsToRegex(tldsFile.data);
+    const matches = tldRegex.match(/[\u0100-\uFFFF]/g) || [];
```
```suggestion
    const matches = tldRegex.match(/[\u007F-\uFFFF]/g) || [];
```

diegocr commented on scripts/update-tld-regex.ts:
```diff
@@ -16,7 +16,15 @@ export async function updateTldRegex() {
+    let tldRegex = domainsToRegex(tldsFile.data);
+    const matches = tldRegex.match(/[\u0100-\uFFFF]/g) || [];
+    matches.forEach(match => {
+        let codePoint = (match.codePointAt(0) || 0).toString(16);
+        if (codePoint.length < 4) {
+            codePoint = `0${codePoint}`;
+        }
```
```suggestion
        const codePoint = (match.codePointAt(0) | 0).toString(16).padStart(4, '0');
```

diegocr committed:
Update scripts/update-tld-regex.ts

diegocr committed:
Update scripts/update-tld-regex.ts

diegocr commented:
Overall looks good, will take care myself from there, thanks!
